editor/SCsub

```diff
@@ -32,13 +32,18 @@ if env["tools"]:
     reg_exporters = "void register_exporters() {\n"
     for e in env.platform_exporters:
         # Glob all .cpp files in export folder
-        files = Glob("#platform/" + e + "/export/" + "*.cpp")
-        env.add_source_files(env.editor_sources, files)
+        if not env["scu_build"]:
+            files = Glob("#platform/" + e + "/export/" + "*.cpp")
+            env.add_source_files(env.editor_sources, files)
 
         reg_exporters += "\tregister_" + e + "_exporter();\n"
         reg_exporters_inc += '#include "platform/' + e + '/export/export.h"\n'
     reg_exporters += "}\n"
 
+    # If we are in a SCU build, we add all the platform exporters in one SCU file.
+    if env["scu_build"]:
+        env.add_source_files(env.editor_sources, "../platform/*.cpp")
```

Comment:
Should be `platform/<name>/export/*.cpp`, no?

This didn't seem needed in the `master` version, what's the difference here?

Reply:
Slight difference here:
In master it compiles each export platform individually. In 3.x, particularly as the export platforms are fairly finalized, these are amalgamated as a single SCU file which is stored in `platform/.scu/` folder. (The path is correct, but it will be confusing as this is the first time you have seen this pattern.)

This makes more sense when read with the `scu_builders.py` file for this:
```
    process_folder(
        [
            "platform",
            "android/export",
            "iphone/export",
            "javascript/export",
            "osx/export",
            "uwp/export",
            "windows/export",
            "x11/export",
        ]
    )
```
This is adding all the various platform exporter folders all into a single uber-SCU file which is stored off the `platform` folder.

_To explain further, the array argument, the first string is the "main" folder, and the following folders are considered sub-folders to be added to the same SCU file._

I'm happy to simplify if preferred to the same method as master, but will lose out on acceleration, but it is worth bearing in mind the exporters are quite significant in the incremental rebuilds. In 3.x, each export directory typically only contains a single file, so there is no accelerating _unless_ they are amalgamated.